src/components/PopupWithForm.js

```diff
@@ -33,4 +34,14 @@ export default class PopupWithForm extends Popup {
     });
     super.setEventListeners();
   }
+
+  loadingButtonState() {
+    this._confirmBtn = this._popupElement.querySelector(".modal__button");
+
+    this._confirmBtn.textContent = "Saving...";
+  }
+
+  resetButtonState() {
+    this._confirmBtn.textContent = `${this._submitType}`;
```

Comment:
`this._confirmBtn` doesn't exist in your class. If you call `resetButtonState` before `loadingButtonState` then your app will crash

So, the best variant is to find the button in line 6 only once